solutions/155. Min Stack.java

```diff
@@ -0,0 +1,50 @@
+class MinStack {
+    
+    private Stack<Integer> base;
+    private Stack<Integer> minStatus;
+    int min;
+​
+    /** initialize your data structure here. */
+    public MinStack() {
+        base=new Stack<Integer>();
+        minStatus=new Stack<Integer>();
+        min=Integer.MAX_VALUE;
+    }
+    
+    public void push(int val) {
+        base.push(val);
+        if(min>val){
+            min=val;
+            minStatus.push(val);
+        }else{
+            minStatus.push(min);
+        }
+    }
+    
+    public void pop() {
+        base.pop();
+        minStatus.pop();
+        if(!minStatus.isEmpty()){
+            min=minStatus.peek();
+        }else{
+            min=Integer.MAX_VALUE;
+        }
+    }
+    
+    public int top() {
+       return base.peek();
+    }
+    
+    public int getMin() {
+        return min;
+    }
+}
+​
```

Comment:
include all the comments and complexities inside the code.
Don't comment on the commit. Creates an issue with PR status internally.